src/main/java/com/trustedchoice/askkodiak/v2/client/AskKodiak.java

```diff
@@ -105,6 +108,82 @@
+     * response. Each product in the results is extended with a property representing the percentage of codes in the
+     * requested group for which the product is eligible.
+     *
+     * @param code  Any valid 2017 NAICS Code (at any level of the NAICS hierarchy) or MD5 Hash (6 Digit NAICS National
+     *              Industry Code + Sub-Description). See above for a more detailed description of possible values.
+     *              Conditional rules which pertain to the specified code will be applied to the results.
+     * @param query Query parameters
+     * @return Eligible products
+     * @throws AskKodiakException error
+     */
+    default Products getProductsEligibleForCode(@Param("code") String code, @QueryMap EligibleQuery query) throws AskKodiakException {
+
+        Map<String, Object> queryParameters = new HashMap<>();
+
+        queryParameters.put("owners", query.getOwners());
+        queryParameters.put("companyType", query.getCompanyType());
+        queryParameters.put("interestLevels", query.getInterestLevels());
+        queryParameters.put("includeEligibility", query.getIncludeEligibility());
+        queryParameters.put("summaryOnly", query.getSummaryOnly());
+        queryParameters.put("productsPerPage", query.getProductsPerPage());
+        queryParameters.put("page", query.getPage());
+        queryParameters.put("geos", query.getGeos());
+        queryParameters.put("productCodes", query.getProductCodes());
+        queryParameters.put("entityTypes", query.getEntityTypes());
+        queryParameters.put("tags", query.getTags());
+        queryParameters.put("annualPayroll", query.getAnnualPayroll());
+        queryParameters.put("annualRevenue", query.getAnnualRevenue());
+        queryParameters.put("anticipatedPremium", query.getAnticipatedPremium());
+        queryParameters.put("fullTimeEmployees", query.getFullTimeEmployees());
+        queryParameters.put("partTimeEmployees", query.getPartTimeEmployees());
+        queryParameters.put("tiv", query.getTiv());
+        queryParameters.put("vehicles", query.getVehicles());
+        queryParameters.put("locations", query.getLocations());
+        queryParameters.put("buildings", query.getBuildings());
+        queryParameters.put("squareFootage", query.getSquareFootage());
+        queryParameters.put("buildingAge", query.getBuildingAge());
+        queryParameters.put("yearsInBusiness", query.getYearsInBusiness());
+        queryParameters.put("yearsInIndustry", query.getYearsInIndustry());
+        queryParameters.put("admitted", query.getAdmitted());
+        queryParameters.put("products", query.getProducts());
+
+        Map<String, Object> queryMap =
+            queryParameters.entrySet()
+                .stream()
+                .filter(e -> e.getValue() != null)
+                .peek(e -> UriUtils.encode(e.getValue().toString()))
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().toString().replace("%2B", "+")));
```

Comment:
This might be a fringe case, but I wonder if there are query params that include a `+` that we want to encode. Does it make sense to encode `+` on the query params that we know (according to the API docs) don't use `+` as a delimiter?